### PR TITLE
[SYCL] Support query of free device memory extension

### PR DESCRIPTION
**Pull request by smaslov-intel**
E2E test for https://github.com/intel/llvm/pull/6604
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/level_zero_device_free_mem.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER% ZES_ENABLE_SYSMAN=1 t.out 2>&1 %GPU_CHECK_PLACEHOLDER
```
Comment (smaslov-intel) [inline SYCL/Plugin/level_zero_device_free_mem.cpp]:
```suggestion
// RUN: env  ZES_ENABLE_SYSMAN=1 %GPU_RUN_PLACEHOLDER t.out 2>&1 %GPU_CHECK_PLACEHOLDER

```

File: SYCL/Plugin/level_zero_device_free_mem.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: env  ZES_ENABLE_SYSMAN=1 %GPU_RUN_PLACEHOLDER t.out 2>&1 %GPU_CHECK_PLACEHOLDER
```
Comment (smaslov-intel) [inline SYCL/Plugin/level_zero_device_free_mem.cpp]:
```suggestion
// RUN: env  ZES_ENABLE_SYSMAN=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER

```

File: SYCL/Plugin/level_zero_device_free_mem.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: env ZES_ENABLE_SYSMAN=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+//
+// The test is to check that the free device memory is reported be Level Zero
+// backend
+//
+// CHECK: Free device memory
```
Comment (againull) [inline SYCL/Plugin/level_zero_device_free_mem.cpp:9]: CHECK will pass only if ext_intel_device_info_free_memory is supported by the device. I think we expect the test to pass even if device doesn't support it. The only option I see is to remove this CHECK, I guess we don't have dynamic checks.

Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_device_free_mem.cpp:9]: But we expect that ALL Level Zero devices support this extension.

Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_device_free_mem.cpp:9]: @againull : are you OK with the test? we need this to complete the new feature (the compiler fix was merged)

**Review by againull — APPROVED** [review]

File: SYCL/Plugin/level_zero_device_free_mem.cpp (hunk 3 above)
Comment (againull) [reply SYCL/Plugin/level_zero_device_free_mem.cpp:9]: Yes, I am ok with the test, thanks for clarification.

**Pull request merged**